packages/marshaler_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import (
+	"context"
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const testFile = "./testdata/marshaler/packages.json"
+
+var generateFlag = flag.Bool("generate", false, "Write golden files")
+
+func TestMarshalJSON(t *testing.T) {
+	// given
+	packagesBasePaths := []string{"../testdata/second_package_path", "../testdata/package"}
+	indexer := NewFileSystemIndexer(packagesBasePaths...)
+	err := indexer.Init(context.Background())
+	require.NoError(t, err, "can't initialize indexer")
+
+	// when
+	m, err := MarshalJSON(&indexer.packageList)
+	require.NoError(t, err)
+
+	// then
+	assertExpectedContent(t, testFile, m)
+}
+
+func TestUnmarshalJSON(t *testing.T) {
+	// given
+	packagesBasePaths := []string{"../testdata/second_package_path", "../testdata/package"}
+	indexer := NewFileSystemIndexer(packagesBasePaths...)
+	err := indexer.Init(context.Background())
+
+	expectedFile, err := os.ReadFile(testFile)
+	require.NoError(t, err)
+
+	var packages Packages
+
+	// when
+	err = UnmarshalJSON(expectedFile, &packages,
+		ResolveBasePaths(packagesBasePaths...),
+		UseFsBuilder(ExtractedFileSystemBuilder))
```

Comment:
Is a FS builder needed here?

Could we have a test unmarshaling a set of packages read from a search index?

I could imagine something like this:
 ```suggestion
	err = UnmarshalJSON(indexSearchFile, &packages, URLPaths(eprBaseURL))
```

Something like `URLPaths` would be only needed if we set the paths in the packages to the URL where the resource is available in the remote storage.

This can perfectly fit in a follow up, but just in case you expect any problem by now.